### PR TITLE
Add accurate type definitions

### PR DESCRIPTION
**Pull request by kcirtaptrick**
Added type definitions, some of which mirror lottie-web
Also resolves microbundle issue with [this comment](https://github.com/mifi/react-lottie-player/pull/23#pullrequestreview-648437316)

Resolves issue: #5, Previous unmerged PRs: #9, #23

### CODE REVIEW

**Comment by jamiehaywood** [comment]
@mifi - having reviewed all 3 PRs, this is the best PR to merge. I think you should merge this as a starting point and then accept PRs to this file moving forward. cc: @skoruba @gugahoi

File: .eslintignore [hunk 1]
```diff
@@ -2,4 +2,5 @@ build/
 dist/
 node_modules/
 .snapshots/
-*.min.js
\ No newline at end of file
+*.min.js
+index.d.ts
```
Comment (jamiehaywood) [inline .eslintignore:6]:
```suggestion
*.d.ts
```

File: package.json [hunk 2]
```diff
@@ -11,8 +11,9 @@
   "engines": {
     "node": ">=10"
   },
+  "types": "dist/index.d.ts",
   "scripts": {
-    "build": "microbundle-crl --no-compress --format modern,cjs",
+    "build": "microbundle-crl --no-compress --format modern,cjs && cp src/index.d.ts dist",
```
Comment (jamiehaywood) [inline package.json:16]: @mifi - I know you'll be publishing this manually - do you have a Windows or MacOS machine? `cp` will only work on windows, so we may need to look into a dev pkg that can execute `copy` on win & MacOS

**Review by jamiehaywood — APPROVED** [review]
two suggestions, but types look tight!

File: package.json (hunk 2 above)
Comment (mifi) [reply package.json:16]: I'm not sure why cp will only work on windows. MacOs has the cp command too. I use macos so it should be ok, we can improve this in the future if needed. pr welcome

**Comment by mifi** [comment]
thanks @kcirtaptrick and @jamiehaywood

File: package.json (hunk 2 above)
Comment (jamiehaywood) [reply package.json:16]: sorry, my apologies a typo- the `cp` command will only work on linux / unix.

**Pull request merged**